In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('data/train.csv', sep=',', encoding='UTF-8')
df1 = df[df.columns[1:29]]
df1.head()

MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0          60       RL         65.0     8450   Pave   NaN      Reg   
1          20       RL         80.0     9600   Pave   NaN      Reg   
2          60       RL         68.0    11250   Pave   NaN      IR1   
3          70       RL         60.0     9550   Pave   NaN      IR1   
4          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities LotConfig    ...     YearBuilt YearRemodAdd RoofStyle  \
0         Lvl    AllPub    Inside    ...          2003         2003     Gable   
1         Lvl    AllPub       FR2    ...          1976         1976     Gable   
2         Lvl    AllPub    Inside    ...          2001         2002     Gable   
3         Lvl    AllPub    Corner    ...          1915         1970     Gable   
4         Lvl    AllPub       FR2    ...          2000         2000     Gable   

  RoofMatl Exterior1st Exterior2nd  MasVnrType  MasVnrArea  ExterQual  \
0  CompShg     VinylSd     VinylSd     BrkFace       196.0         Gd   
1  CompShg     MetalSd     MetalSd        None         0.0         TA   
2  CompShg     VinylSd     VinylSd     BrkFace       162.0         Gd   
3  CompShg     Wd Sdng     Wd Shng        None         0.0         TA   
4  CompShg     VinylSd     VinylSd     BrkFace       350.0         Gd   

   ExterCond  
0         TA  
1         TA  
2         TA  
3         TA  
4         TA  

[5 rows x 28 columns]

In [3]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 28 columns):
MSSubClass      1460 non-null int64
MSZoning        1460 non-null object
LotFrontage     1201 non-null float64
LotArea         1460 non-null int64
Street          1460 non-null object
Alley           91 non-null object
LotShape        1460 non-null object
LandContour     1460 non-null object
Utilities       1460 non-null object
LotConfig       1460 non-null object
LandSlope       1460 non-null object
Neighborhood    1460 non-null object
Condition1      1460 non-null object
Condition2      1460 non-null object
BldgType        1460 non-null object
HouseStyle      1460 non-null object
OverallQual     1460 non-null int64
OverallCond     1460 non-null int64
YearBuilt       1460 non-null int64
YearRemodAdd    1460 non-null int64
RoofStyle       1460 non-null object
RoofMatl        1460 non-null object
Exterior1st     1460 non-null object
Exterior2nd     1460 non-null object
MasVnrType  

### features info  
  
- MSSubClass      1460 non-null int64 / categorical  
- MSZoning        1460 non-null object / categorical  
- LotFrontage     1201 non-null float64  
- LotArea         1460 non-null int64  
- Street          1460 non-null object / ordinal  
- Alley           91 non-null object / ordinal / NA = 0  
- LotShape        1460 non-null object / ordinal  
- LandContour     1460 non-null object / categorical  
- Utilities       1460 non-null object / ordinal  
- LotConfig       1460 non-null object / categorical  
- LandSlope       1460 non-null object / ordinal  
- Neighborhood    1460 non-null object / categorical  
- Condition1      1460 non-null object / categorical  
- Condition2      1460 non-null object / categorical  
- BldgType        1460 non-null object / categorical  
- HouseStyle      1460 non-null object / categorical  
- OverallQual     1460 non-null int64 / ordinal  
- OverallCond     1460 non-null int64 / ordinal  
- YearBuilt       1460 non-null int64  
- YearRemodAdd    1460 non-null int64  
- RoofStyle       1460 non-null object / categorical  
- RoofMatl        1460 non-null object / categorical  
- Exterior1st     1460 non-null object / categorical  
- Exterior2nd     1460 non-null object / categorical  
- MasVnrType      1452 non-null object / categorical  
- MasVnrArea      1452 non-null float64  
- ExterQual       1460 non-null object / ordinal  
- ExterCond       1460 non-null object / categorical

In [4]:
cols_num = ['LotFrontage', 'LotArea', 'YearBuilt', 'YearRemodAdd', 'MasVnrArea']
cols_ord = ['Street', 'Alley', 'LotShape', 'Utilities', 'LandSlope', 'OverallQual', 'OverallCond', 'ExterQual']
cols_cat = ['MSSubClass', 'MSZoning', 'LandContour', 'LotConfig', 'Neighborhood', 'Condition1', 'Condition2',
            'BldgType', 'HouseStyle', 'RoofStyle', 'RoofMatl', 'Exterior1st', 'Exterior2nd', 'MasVnrType', 'ExterCond']

In [5]:
df1[cols_num].info()
df1[cols_ord].info()
df1[cols_cat].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 5 columns):
LotFrontage     1201 non-null float64
LotArea         1460 non-null int64
YearBuilt       1460 non-null int64
YearRemodAdd    1460 non-null int64
MasVnrArea      1452 non-null float64
dtypes: float64(2), int64(3)
memory usage: 57.1 KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 8 columns):
Street         1460 non-null object
Alley          91 non-null object
LotShape       1460 non-null object
Utilities      1460 non-null object
LandSlope      1460 non-null object
OverallQual    1460 non-null int64
OverallCond    1460 non-null int64
ExterQual      1460 non-null object
dtypes: int64(2), object(6)
memory usage: 91.3+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 15 columns):
MSSubClass      1460 non-null int64
MSZoning        1460 non-null object
LandContour     1460 non-null object
L

In [6]:
df1.isnull().sum()

MSSubClass         0
MSZoning           0
LotFrontage      259
LotArea            0
Street             0
Alley           1369
LotShape           0
LandContour        0
Utilities          0
LotConfig          0
LandSlope          0
Neighborhood       0
Condition1         0
Condition2         0
BldgType           0
HouseStyle         0
OverallQual        0
OverallCond        0
YearBuilt          0
YearRemodAdd       0
RoofStyle          0
RoofMatl           0
Exterior1st        0
Exterior2nd        0
MasVnrType         8
MasVnrArea         8
ExterQual          0
ExterCond          0
dtype: int64

### 결측값 (nan) 처리
    - numerical (수치형) : LotFrontage(거리에서 집까지 발걸음), MasVnrArea(벽돌장식 면적)  
    - ordinal (서수형) : Alley(골목길 접근성)  
    - categorical (카테고리형) : MasVnrType(벽돌장식 타입)

In [7]:
df1[cols_num].head()

LotFrontage  LotArea  YearBuilt  YearRemodAdd  MasVnrArea
0         65.0     8450       2003          2003       196.0
1         80.0     9600       1976          1976         0.0
2         68.0    11250       2001          2002       162.0
3         60.0     9550       1915          1970         0.0
4         84.0    14260       2000          2000       350.0

In [8]:
# 수치형 결측값은 해당 컬럼의 평균값 입력
df1[cols_num[0]] = df1[cols_num[0]].fillna(df1['LotFrontage'].mean())
df1[cols_num[4]] = df1[cols_num[4]].fillna(df1['MasVnrArea'].mean())

C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  
C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [9]:
df1[cols_ord].head()

Street Alley LotShape Utilities LandSlope  OverallQual  OverallCond  \
0   Pave   NaN      Reg    AllPub       Gtl            7            5   
1   Pave   NaN      Reg    AllPub       Gtl            6            8   
2   Pave   NaN      IR1    AllPub       Gtl            7            5   
3   Pave   NaN      IR1    AllPub       Gtl            7            5   
4   Pave   NaN      IR1    AllPub       Gtl            8            5   

  ExterQual  
0        Gd  
1        TA  
2        Gd  
3        TA  
4        Gd

In [10]:
# 서수형 결측값은 data description에 따라 NA = 0인 경우 0으로 입력
df1[cols_ord[1]] = df1[cols_ord[1]].fillna(0)

C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [11]:
df1[cols_cat].head()

MSSubClass MSZoning LandContour LotConfig Neighborhood Condition1  \
0          60       RL         Lvl    Inside      CollgCr       Norm   
1          20       RL         Lvl       FR2      Veenker      Feedr   
2          60       RL         Lvl    Inside      CollgCr       Norm   
3          70       RL         Lvl    Corner      Crawfor       Norm   
4          60       RL         Lvl       FR2      NoRidge       Norm   

  Condition2 BldgType HouseStyle RoofStyle RoofMatl Exterior1st Exterior2nd  \
0       Norm     1Fam     2Story     Gable  CompShg     VinylSd     VinylSd   
1       Norm     1Fam     1Story     Gable  CompShg     MetalSd     MetalSd   
2       Norm     1Fam     2Story     Gable  CompShg     VinylSd     VinylSd   
3       Norm     1Fam     2Story     Gable  CompShg     Wd Sdng     Wd Shng   
4       Norm     1Fam     2Story     Gable  CompShg     VinylSd     VinylSd   

  MasVnrType ExterCond  
0    BrkFace        TA  
1       None        TA  
2    BrkFace        TA  
3       None        TA  
4    BrkFace        TA

In [12]:
# 카테고리형 결측값은 하나의 unique 값으로 판단하여 0으로 입력
df1[cols_cat[-2]] = df1[cols_cat[-2]].fillna(0)

C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [13]:
df1.isnull().sum()

MSSubClass      0
MSZoning        0
LotFrontage     0
LotArea         0
Street          0
Alley           0
LotShape        0
LandContour     0
Utilities       0
LotConfig       0
LandSlope       0
Neighborhood    0
Condition1      0
Condition2      0
BldgType        0
HouseStyle      0
OverallQual     0
OverallCond     0
YearBuilt       0
YearRemodAdd    0
RoofStyle       0
RoofMatl        0
Exterior1st     0
Exterior2nd     0
MasVnrType      0
MasVnrArea      0
ExterQual       0
ExterCond       0
dtype: int64

### 수치화

In [14]:
# 서수형

tmp_ord = df1[['Street', 'Alley']]
for col in tmp_ord:
    df1[col][df1[col] == 'Grvl'] = (1 - 1/2) / 2
    df1[col][df1[col] == 'Pave'] = (2 - 1/2) / 2
    
    print(df1[col].unique())

C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """
C:\Python\Anaconda3-5.2\lib\site-packages\pandas\core\generic.py:7620: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)
C:\Python\Anaconda3-5.2\lib\site-packages\IPython\core\interactiveshell.py:2963: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  exec(code_obj, self.user_global_ns, self.user_ns)
C:\Python\Anaconda3-5.2\lib\site-packages\ipyker

[0.75 0.25]
[0 0.25 0.75]


In [15]:
df1['LotShape'].unique()

df1['LotShape'][df1['LotShape'] == 'Reg'] = (4 - 1/2) / 4
df1['LotShape'][df1['LotShape'] == 'IR1'] = (3 - 1/2) / 4
df1['LotShape'][df1['LotShape'] == 'IR2'] = (2 - 1/2) / 4
df1['LotShape'][df1['LotShape'] == 'IR3'] = (1 - 1/2) / 4

df1['LotShape'].unique()

C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until
C:\Python\Anaconda3-5.2\lib\site-packages\pandas\core\generic.py:7620: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)
C:\Python\Anaconda3-5.2\lib\site-packages\IPython\core\interactiveshell.py:2963: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  exec(code_obj, self.us

array([0.875, 0.625, 0.375, 0.125], dtype=object)

In [16]:
df1['Utilities'].unique()

df1['Utilities'][df1['Utilities'] == 'AllPub'] = (4 - 1/2) / 4
df1['Utilities'][df1['Utilities'] == 'NoSewr'] = (3 - 1/2) / 4
df1['Utilities'][df1['Utilities'] == 'NoSeWa'] = (2 - 1/2) / 4
df1['Utilities'][df1['Utilities'] == 'ELO'] = (1 - 1/2) / 4

df1['Utilities'].unique()

C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until
C:\Python\Anaconda3-5.2\lib\site-packages\pandas\core\generic.py:7620: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)
C:\Python\Anaconda3-5.2\lib\site-packages\IPython\core\interactiveshell.py:2963: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  exec(code_obj, self.us

array([0.875, 0.375], dtype=object)

In [17]:
df1['LandSlope'].unique()

df1['LandSlope'][df1['LandSlope'] == 'Gtl'] = (3 - 1/2) / 3
df1['LandSlope'][df1['LandSlope'] == 'Mod'] = (2 - 1/2) / 3
df1['LandSlope'][df1['LandSlope'] == 'Sev'] = (1 - 1/2) / 3

df1['LandSlope'].unique()

C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until
C:\Python\Anaconda3-5.2\lib\site-packages\pandas\core\generic.py:7620: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)
C:\Python\Anaconda3-5.2\lib\site-packages\IPython\core\interactiveshell.py:2963: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  exec(code_obj, self.us

array([0.8333333333333334, 0.5, 0.16666666666666666], dtype=object)

In [18]:
df1['ExterQual'].unique()

df1['ExterQual'][df1['ExterQual'] == 'Ex'] = (5 - 1/2) / 5
df1['ExterQual'][df1['ExterQual'] == 'Gd'] = (4 - 1/2) / 5
df1['ExterQual'][df1['ExterQual'] == 'TA'] = (3 - 1/2) / 5
df1['ExterQual'][df1['ExterQual'] == 'Fa'] = (2 - 1/2) / 5
df1['ExterQual'][df1['ExterQual'] == 'Po'] = (1 - 1/2) / 5

df1['ExterQual'].unique()

C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until
C:\Python\Anaconda3-5.2\lib\site-packages\pandas\core\generic.py:7620: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)
C:\Python\Anaconda3-5.2\lib\site-packages\IPython\core\interactiveshell.py:2963: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  exec(code_obj, self.us

array([0.7, 0.5, 0.9, 0.3], dtype=object)

In [19]:
# 카테고리형

for col in df1[cols_cat]:
    val = 1
    tmp_dict = {}
    
    for unique in df1[col].unique():
        
        if unique != 0:
            tmp_dict[unique] = val
            df1[col][df1[col] == unique] = val
            val += 1
            
    print(tmp_dict)

C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  # This is added back by InteractiveShellApp.init_path()
C:\Python\Anaconda3-5.2\lib\site-packages\pandas\core\generic.py:7620: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)
C:\Python\Anaconda3-5.2\lib\site-packages\IPython\core\interactiveshell.py:2963: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  exec(code_obj, self.user_global_ns, self.user

{60: 1, 20: 2, 70: 3, 50: 4, 190: 5, 45: 6, 90: 7, 120: 8, 30: 9, 85: 10, 80: 11, 160: 12, 75: 13, 180: 14, 40: 15}
{'RL': 1, 'RM': 2, 'C (all)': 3, 'FV': 4, 'RH': 5}
{'Lvl': 1, 'Bnk': 2, 'Low': 3, 'HLS': 4}
{'Inside': 1, 'FR2': 2, 'Corner': 3, 'CulDSac': 4, 'FR3': 5}
{'CollgCr': 1, 'Veenker': 2, 'Crawfor': 3, 'NoRidge': 4, 'Mitchel': 5, 'Somerst': 6, 'NWAmes': 7, 'OldTown': 8, 'BrkSide': 9, 'Sawyer': 10, 'NridgHt': 11, 'NAmes': 12, 'SawyerW': 13, 'IDOTRR': 14, 'MeadowV': 15, 'Edwards': 16, 'Timber': 17, 'Gilbert': 18, 'StoneBr': 19, 'ClearCr': 20, 'NPkVill': 21, 'Blmngtn': 22, 'BrDale': 23, 'SWISU': 24, 'Blueste': 25}
{'Norm': 1, 'Feedr': 2, 'PosN': 3, 'Artery': 4, 'RRAe': 5, 'RRNn': 6, 'RRAn': 7, 'PosA': 8, 'RRNe': 9}
{'Norm': 1, 'Artery': 2, 'RRNn': 3, 'Feedr': 4, 'PosN': 5, 'PosA': 6, 'RRAn': 7, 'RRAe': 8}
{'1Fam': 1, '2fmCon': 2, 'Duplex': 3, 'TwnhsE': 4, 'Twnhs': 5}
{'2Story': 1, '1Story': 2, '1.5Fin': 3, '1.5Unf': 4, 'SFoyer': 5, 'SLvl': 6, '2.5Unf': 7, '2.5Fin': 8}
{'Gable': 1,

In [20]:
df1[cols_cat].head()

MSSubClass MSZoning LandContour LotConfig Neighborhood Condition1  \
0           1        1           1         1            1          1   
1           2        1           1         2            2          2   
2           1        1           1         1            1          1   
3           3        1           1         3            3          1   
4           1        1           1         2            4          1   

  Condition2 BldgType HouseStyle RoofStyle RoofMatl Exterior1st Exterior2nd  \
0          1        1          1         1        1           1           1   
1          1        1          2         1        1           2           2   
2          1        1          1         1        1           1           1   
3          1        1          1         1        1           3           3   
4          1        1          1         1        1           1           1   

  MasVnrType ExterCond  
0          1         1  
1          2         1  
2          1         1  
3          2         1  
4          1         1

### 표준화
    Standardization : (요소값 - 평균) / 표준편차
    Normalization : (요소값 - 최소값) / (최대값 - 최소값)

In [21]:
# 수치형

for col in cols_num:
    df1[col] = (df1[col] - df1[col].mean()) / df1[col].std()

df1[cols_num].head()

C:\Python\Anaconda3-5.2\lib\site-packages\ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  after removing the cwd from sys.path.


LotFrontage   LotArea  YearBuilt  YearRemodAdd  MasVnrArea
0    -0.229293 -0.207071   1.050634      0.878367    0.511243
1     0.451781 -0.091855   0.156680     -0.429430   -0.574214
2    -0.093078  0.073455   0.984415      0.829930    0.322950
3    -0.456318 -0.096864  -1.862993     -0.720051   -0.574214
4     0.633401  0.375020   0.951306      0.733056    1.364102

In [22]:
df1.to_csv('./data/HousePrice_preprocessing_df1', sep = ',', encoding = 'UTF-8')